src/build/cargo.rs

```diff
@@ -315,11 +323,26 @@ impl Executor for RlsExecutor {
         // with the same crate name but different hashes, e.g., those
         // made with a different compiler.
         let cargo_args = cargo_cmd.get_args();
+
+        // map these ones and for all since we look at it several times.
+        let cargo_args_str:Vec<&str> = cargo_args.iter().map(|c| c.to_str().unwrap_or("")).collect();
+
         let crate_name =
-            parse_arg(cargo_args, "--crate-name").expect("no crate-name in rustc command line");
-        trace!("exec: {}", crate_name);
+            parse_arg(&cargo_args_str, "--crate-name").expect("no crate-name in rustc command line");
+        let target_name =
+            parse_target(&cargo_args_str).expect("no compile target in rustc command line");
+        trace!("exec: {} {}", crate_name, target_name);
+
+        // Send off a window/progress notification for this compile target.
+        // At the moment, we don't know the number of things cargo is going to compile,
+        // so we just send the name of each thing we find.
+        {
+            let progress_sender = self.progress_sender.lock().unwrap();
+            progress_sender.send(ProgressUpdate::Message(format!("{} {}", crate_name, target_name)))
+                .expect("Failed to send progress update");
+        }
```

Comment:
Progress in the first (non-cached) build sends messages on the form `<crate> <source>`, i.e. `infer_lib src/lib.rs`. Is this a good idea?

Reply:
I think just the crate name should be enough - the source path is usually obvious from the crate